hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/AltoChannelConnector.java

```diff
@@ -176,6 +176,9 @@ private void onFailure(Channel channel) {
 
             failed = true;
             closeAllChannels();
+            logger.warning("Alto channel establishments for the " + connection + " have failed. "
+                    + "The client will not be using the Alto channels to route partition specific invocations, "
+                    + "and fallback to the smart routing mode for this connection.");
```

Comment:
Add a small note about firewall settings. That happened in one of our benchmarks (ec2 security group not configured correctly).

Reply:
https://github.com/hazelcast/hazelcast/pull/23302/commits/6a8433ff694c5c2429361b07c10bc4fb02067572